src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -64,6 +71,27 @@ void ClusterTaskManager::QueueAndScheduleTask(
   ScheduleAndDispatchTasks();
 }
 
+void ClusterTaskManager::QueueAndScheduleTask(
```

Comment:
similarly, `forward_to` and `use_required_resources` should not be part of `QueueAndScheduleTask` API. 
Additionally, we should only have one `QueueAndScheduleTask` function

Reply:
Ditto.

Reply:
Yea, we should only have the original QueueAndScheduleTask? Raylet based scheduling should directly talk to the remote cluster_task_manager?

The way how I see this is that gcs is just a node like raylet (it can have a special node id). Gcs actor scheduler is just the lease client and it will send the lease request to an initial node (can be gcs node or raylet node based on config) and then handle spillback, rejection accordingly. The difference is that gcs node will only spillback but that should be transparent to the lease client.